src/types.rs

```diff
@@ -246,14 +243,33 @@ pub struct RecordedEvent {
     /// Representing the user-defined metadata associated with this event.
     pub custom_metadata: Bytes,
 
-    /// Indicates wheter the content is internally marked as JSON.
-    pub is_json: bool,
-
     /// An event position in the $all stream.
     pub position: Position,
 }
 
 impl RecordedEvent {
+    /// Type of this event.
+    pub fn event_type(&self) -> &str {
+        if let Some(tpe) = self.metadata.get("type") {
+            &tpe
+        } else {
+            "<no-event-type-provided>"
+        }
+    }
+
+    /// Indicates whether the content is internally marked as JSON.
+    pub fn is_json(&self) -> bool {
```

Comment:
Considering that property is frequently computed, I would also keep it cached in the struct itself.

Reply:
@YoEight reverted